community/cypher/cypher/src/main/scala/org/neo4j/cypher/internal/compatibility/v3_3/runtime/executionplan/procs/ProcedureExecutionResult.scala

```diff
@@ -95,9 +95,8 @@ class ProcedureExecutionResult(context: QueryContext,
   override def accept[EX <: Exception](visitor: QueryResultVisitor[EX]): Unit = {
     executionResults.foreach { res =>
       val fieldArray = new Array[AnyValue](indexResultNameMappings.size)
-      var i = 0
-      for ((pos, _, typ) <- indexResultNameMappings) {
-        fieldArray(i) = typ match {
+      for (pos <- indexResultNameMappings.indices) {
+        fieldArray(pos) = indexResultNameMappings(pos)._3 match {
```

Comment:
is this a bottleneck? If not I think the old way looks nicer, not a fan of `._3`

Reply:
Amazingly the profiling showed this to be 2x faster for this part of the profile. It turned out that the call stack created by scala for the tuple foreach was much, much deeper than the one for the int range foreach. No idea why, but the performance change was very real.

So we did not do this whimsically.